lib/src/main/kotlin/io/nexure/fsm/CountryStateMachineInterfaces.kt

```diff
@@ -0,0 +1,20 @@
+package io.nexure.fsm
+
+import com.neovisionaries.i18n.CountryCode
+
+interface CountryStateMachineConfig<S : Any, E : Any> {
+    val country: CountryCode
+    val enabled: Boolean
+    val transitions: List<CountryEdge<S, E>>
+    val initialState: S
+}
```

Comment:
this library is extremely generic, so I would avoid introducing references to the business logic. Instead, I’d suggest introducing something like a `descriptor` field to characterize the state machine configuration

```suggestion
interface StateMachineConfig<D: Any, S : Any, E : Any> {
    val descriptor: D
    val enabled: Boolean
    val transitions: List<Edge<S, E>>
    val initialState: S
}
```